Button component

ericvalcik opened the pull request:
This PR should resolve [this issue](https://github.com/webscopeio/mailingui-web/issues/91).

Please let me know if you have any comments @jvorcak @iamhectorsosa. I still need to test the examples on the email clients, so please don't merge yet, but a review is appreciated.

Before merging this PR we should merge the [Badges PR](https://github.com/webscopeio/mailingui-web/pull/90) because I created this branch directly out of the `badge-improvements` branch in the mentioned PR.

iamhectorsosa commented on src/components/ComponentExample/ComponentExample.tsx:
```diff
@@ -139,3 +182,10 @@ function transformComponentName(componentName: string): string {
   return transformedName;
 }
+
+// Mappings of component name to iframe height
+const getIframeHeight = (type: string): number => {
+  if (type === "badges") return 120;
+  if (type === "buttons") return 140;
+  return 350;
+};
```
This could either be an object or a Switch function.

iamhectorsosa commented on src/mailingui/components/buttons/Buttons.tsx:
```diff
@@ -0,0 +1,48 @@
+import React, { FC, ReactNode, CSSProperties } from "react";
+import { Button as ReactEmailButton } from "@react-email/components";
+import { Property } from "csstype";
+
+interface ButtonProps {
+  paddingX?: number;
+  paddingY?: number;
+  href: string;
+  children: ReactNode;
+  backgroundColor?: Property.Color;
+  fontSize?: Property.FontSize<string | number>;
+  color?: Property.Color;
+  borderColor?: Property.Color;
+  pill?: boolean;
+}
```
You are already using `React.CSSProperties`, you can extend all of those types as well for your interface (i.e.,`backgroundColor: CSSProperties['color']`) and it should work.

Any reason why using types from `csstype`?

jvorcak commented on src/mailingui/components/buttons/Buttons.tsx:
```diff
@@ -0,0 +1,47 @@
+  fontSize?: CSSProperties["fontSize"];
+  color?: CSSProperties["color"];
+  borderColor?: CSSProperties["color"];
+  pill?: boolean;
```
I'm not quite sure whether I like this way of having `pill` as a property. 

What do you guys think if we instead provide some enum for `rounded` property?

1. I had to google what do we mean by `pill`, it was not very obvious
2. If we want to have let's say 3 versions of how rounded borders we will have, it won't be possible.

WDYT @iamhectorsosa @ericvalcik ?

ericvalcik replied:
I think both can work. I have no problem with changing it to `rounded` with an enum with values like `none`, `sm`, `md`, `lg` `full` for example.

I agree with the proposal

iamhectorsosa replied:
I like the change @ericvalcik did

ericvalcik commented:
I fixed the mentioned issues and implemented the proposals.

But when testing I found out these buttons are not getting rendered at all in Outlook clients and it is because of the react.email Button component. So in the last commit, I copied the component from react.email and modified it so it is getting rendered in Outlook.

I tested in testi.at and even though ugly, the buttons are being rendered in the Outlook clients. Every other client renders the components the same way we show them on our web.

I think this should be ready to merge now, check it out @jvorcak @iamhectorsosa

jvorcak commented on src/mailingui/components/buttons/Buttons.tsx:
```diff
@@ -0,0 +1,89 @@
+  backgroundColor?: CSSProperties["color"];
+  color?: CSSProperties["color"];
+  borderColor?: CSSProperties["color"];
+  rounded?: "none" | "sm" | "md" | "lg" | "full";
```
I'm thinking, should we make this just a plain integer? (overridable with a template with a reasonable default?)

jvorcak commented on src/mailingui/components/buttons/ReactEmailButtonFork.tsx:
```diff
@@ -0,0 +1,57 @@
+import * as React from "react";
```
I get why we need this, but the naming seems pretty weird to me to be honest. @iamhectorsosa is there any good pattern how we can override these library files until we file a proper PRs there?

iamhectorsosa replied:
Not sure tbh, I don't have major concerns since this is only for internal consumption.

It is a proprietary implementation though, we aren't importing/consuming anything from React Email. We aren't overriding anything either.

So maybe using a name that alludes to its function, something like "ButtonPrimitive" or "ButtonWrapper", would be better? I'm gonna leave this one to @ericvalcik 

Also could we leave a comment in the component with a TODO to replace this if we are really intending to file a PR to React.email.

```tsx
// If we are expecting to replace this, we can keep the named import as is
import { Button as ReactEmailButton } from "./<FILENAME>";
// To be replaced by
import { Button as ReactEmailButton } from "@react-email/components";
```

iamhectorsosa commented on src/mailingui/components/buttons/Button.tsx:
```diff
@@ -0,0 +1,81 @@
+  );
+};
+
+const sizesEnum: Record<
```
I would prefer to call this variant rather than enum, and also play around with a structure like this perhaps:

```tsx
// When I consume, I would like something like this
const style = {
  color: variants.color[color]
  fontSize: variants.fontSize[size]
  paddingInline: variants.padding[size].x
  paddingBlock: variants.padding[size].y
  borderRadius: variants.borderRadius[rounded]
}

// I know it is a bit repetitive, but you are accessing the property and then the variant from that property.

// Like variants[<property>][<variant>]

const variants = {
  color: {
    primary: { //... },
    secondary: { //... },
  },
  fontsize: {
    sm: { //... },
    md: { //... },
    lg: { //... },
  },
  borderRadius: {
    sm: { //... },
    md: { //... },
    lg: { //... },
  }
  //..
}
```

Check CVA
https://cva.style/docs/getting-started#your-first-component

Also TailwindCSS follows a similar convention
https://tailwindcss.com/docs/customizing-spacing#overriding-the-default-spacing-scale

We also have to discuss and see how would this fit with having the themes in place.

ericvalcik replied:
I think we can make these changes in the next PR, where we introduce the ThemeProvider, right?

I'll make a note to not forget to consider this when doing the PR.

iamhectorsosa commented on src/mailingui/components/buttons/ReactEmailButtonFork.tsx:
```diff
@@ -0,0 +1,57 @@
+  pY?: number;
+}
+
+export const Button = React.forwardRef<ButtonElement, Readonly<ButtonProps>>(
```
I'm curious, why would we need this utility for component props? Readonly constructs a type which properties cannot be reassigned. Is this a pattern we need?
